src/parser/core/modules/ActionWindow/windows/BuffWindow.ts

```diff
@@ -50,28 +51,42 @@ export abstract class BuffWindow extends ActionWindow {
 			.map(actor => actor.id)
 
 		const targets = [this.parser.actor.id, ...enemyTargets]
+		const playerOwnedIds = this.parser.pull.actors
+			.filter(actor => (actor.owner === this.parser.actor) || actor === this.parser.actor)
+			.map(actor => actor.id)
 
 		const buffFilter = filter<Event>()
-			.source(this.parser.actor.id)
+			.source(oneOf(playerOwnedIds))
 			.target(oneOf(targets))
 			.status(oneOf(ensureArray(this.buffStatus).map(s => s.id)))
 
-		this.addEventHook(buffFilter.type('statusApply'), this.startWindowAndTimeout)
-		this.addEventHook(buffFilter.type('statusRemove'), this.endWindowByStatus)
+		this.addEventHook(buffFilter.type('statusApply'), this.onStatusApply)
+		this.addEventHook(buffFilter.type('statusRemove'), this.onStatusRemove)
+		this.buffDuration = _.max(ensureArray(this.buffStatus).map(s => s.duration))
```

Comment:
Does this need lodash? Not requesting any change, just wondering if there's a benefit to lodash over normal `Math.max` and spreading the array which seems like it would be more readable.

Reply:
I kept the implementation that already existed in the BuffWindow, I don't have an opinion about whether lodash is necessary.